### PR TITLE
Refactor DOM and remove the items number limitation

### LINKED ISSUE #43: Why hardcode the maximum number of menu items?
## Expected Behavior
Some stores have big menus with 8 or 10 or more items. We should be able to configure what's the maximum.

## Current Behavior
There's a hardcoded limit of 6 items, see https://github.com/vtex-apps/category-menu/blob/9caf949ebeaa6d16f1966d8c9f900f3caf783fbc/react/index.js#L14

## Possible Solution
Put the maximum number of items as a property on the schema as a layout option, or allow sending the maximum value by props.

## Context
The layout the client wants cannot be done in the current state of the app
screenshot at jan 15 14-48-58

### PR DESCRIPTION
**Pull request by rerissondaniel**
#### What is the purpose of this pull request?
Improve the code quality and use proper elements in DOM. Also, it resolves #43 by removing the max number of items in the menu.
#### What problem is this solving?

The app wasn't using `ul` and `li`, even though it was all about lists. Also, this PR makes the code clearer and any suggestions in that sense are welcome.
#### How should this be manually tested?
[workspace](https://useul--storecomponents.myvtex.com/)
#### Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/CategoryItem.js [hunk 1]
```diff
@@ -14,69 +14,77 @@ import categoryMenu from '../categoryMenu.css'
...
-        'b--transparent': !isHover,
-        'b--action-primary': isHover,
+        'b--transparent': !isOnHover,
+        'b--action-primary': isOnHover,
       }
     )
 
-    const itemClasses = `${categoryMenu.container} flex justify-center items-center`
+    return noRedirect ? (
+      <a href="#" className={linkClasses}>
```
Comment (brunojdo) [inline react/components/CategoryItem.js]: Why we use this approach?

Comment (brunojdo) [reply react/components/CategoryItem.js]: seems awkward

Comment (rerissondaniel) [reply react/components/CategoryItem.js]: It's used to render the 'Departments' category. Now its using a span. If we decided to render the departments separately we would have duplicated code for rendering the title and the children.

File: react/components/CategoryItem.js [hunk 2]
```diff
@@ -14,69 +14,77 @@ import categoryMenu from '../categoryMenu.css'
...
+        onCloseMenu={this.handleCloseMenu}
+        showSecondLevel={subcategoryLevels === 2}
+      />
+    )
+  }
+  
+  render() {
     return (
-      <div className={itemClasses}
+      <li className={`${categoryMenu.container, categoryMenu.list} flex items-center db list`}
```
[the remark this answers is not in the document]
Comment (rerissondaniel) [reply react/components/CategoryItem.js]: Yeah, thanks!

**Pull request merged**